src/CoolMQTT.cpp

```diff
@@ -186,46 +193,47 @@ bool CoolMQTT::publish(const char* data)
 	Serial.println();
 #endif
 
-	bool pub=client.publish( this->outTopic,(byte*) data,strlen(data),false  );
-	delay(100);			//wait a little and treat network
-	yield();
 
 #if DEBUG == 1 
 
 	Serial.print( F("success : ") );
 	Serial.println(pub);	
 
 #endif
-	byte repeats = 0;
-	while (pub == 0)
+	byte retries = 0;
+	//bool published = false;
+	bool published = client.publish( this->outTopic,(byte*) data,strlen(data),false  );
+	while (!published && retries < 5)
 	{
-		Serial.println( F("Publish : FAIL!!!"));
-		if (wifiManager.state()!=3)
+		published = client.publish( this->outTopic,(byte*) data,strlen(data),false  );
+		if (!published )
 		{
-			Serial.println( F("No WiFi Re-connecting.."));
-			wifiManager.disconnect();
-			delay(200);
-			wifiManager.begin();
-		}
-		if (state() != 0)
-		{
-			Serial.println( F("No MQTT Re-connecting.."));
-			delay(200);
-			connect(1000);
+			if (wifiManager.state() != 3)
```

Comment:
magic number (`3`). Is there a `#define` for this in `WifiManager.h` ?

Reply:
wifi status is defined in the ESP8266 wifi Client(i guess), for quick reference i copied them in a comment in CoolWifi.status().
WL_CONNECTED        = 3,